Updated building-cmake.md for Ubuntu 14.04+

Helias opened the pull request:
I made this changes in "building-cmake Ubuntu 14.04.md" and updated the link of zlib in builind-gmake.md

What change from building-cmake:

- Converted apt-get into apt
- Added libdrm-dev and libtool on the command tu install libraries
- Updated the branch for the stable version of OpenSSL (as @john-preston suggested in #2519)
- Added some troubleshooting for libicu* libraries (as suggested in #2719 )

I think we can close #2719 with this Pull Request.

Tested on Ubuntu 16.04 and 14.04

nachoparker commented:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  pkbrister@zoho.com
    Domain ownyourbits.com has exceeded the max emails per hour (3/3 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 5150 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:55347 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by skywalker.onesite.co with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1cWXMQ-002Swk-KK
	for nacho@ownyourbits.com; Wed, 25 Jan 2017 18:52:18 -0500
Date: Wed, 25 Jan 2017 15:51:38 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1485388298;
	bh=d09fDz4a+LriOs3Ob6H743TQCVqiaN08nos+ShMkhKU=;
	h=From:Reply-To:To:Cc:Subject:List-ID:List-Archive:List-Post:
	 List-Unsubscribe:From;
	b=bUsrIuGq5DJeS/y52X+WK2ihODMtf/OfdENZIz/KVkOVYgsiHOo7sDWFGjjs+F/m3
	 8Kryt4CBtmSX3+oPkEmKdJ1+F8/r5tSOsuvr3Lu2rnHxRZYHCbatgMHSbCkUumenXj
	 ENqU1FMg/rSESiPQ3iKP0m/FV93oqCAGA213TbYg=
From: Stefano Borzì <notifications@github.com>
Reply-To: telegramdesktop/tdesktop <reply@reply.github.com>
To: telegramdesktop/tdesktop <tdesktop@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Message-ID: <telegramdesktop/tdesktop/pull/2926@github.com>
Subject: [telegramdesktop/tdesktop] Updated building-cmake.md for Ubuntu 16.04
 (#2926)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_58893a0ae10bd_7c6c3ff2f0213130109582";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: Helias
X-GitHub-Recipient: nachoparker
X-GitHub-Reason: subscribed
List-ID: telegramdesktop/tdesktop <tdesktop.telegramdesktop.github.com>
List-Archive: https://github.com/telegramdesktop/tdesktop
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+0145ac5c227886e2739028545d8712794ec49eaf5c735dcc92cf0000000114a0fc0a92a169ce0c1d7629@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AUWsXOuvplx_5fQTSq_V8M0WjOoWsg9fks5rV-AKgaJpZM4LuJXq>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: nacho@ownyourbits.com


----==_mimepart_58893a0ae10bd_7c6c3ff2f0213130109582
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

I made this changes in building-cmake.md

- Converted apt-get into apt
- Updated zlib download link
- Updated the branch for the stable version of OpenSSL (as @john-preston says in #2519)
- Added ppa for libdrm
- Added some troubleshooting for libicu* libraries (as suggested in #2719 )

I think we can Close #2719 with this Pull Request.

Tested in Ubuntu 16.04


You can view, comment on, or merge this pull request online at:

  https://github.com/telegramdesktop/tdesktop/pull/2926

-- Commit Summary --

  * Updated building-cmake.md for Ubuntu 16.04

-- File Changes --

    M doc/building-cmake.md (48)

-- Patch Links --

https://github.com/telegramdesktop/tdesktop/pull/2926.patch
https://github.com/telegramdesktop/tdesktop/pull/2926.diff

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/telegramdesktop/tdesktop/pull/2926

----==_mimepart_58893a0ae10bd_7c6c3ff2f0213130109582
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>I made this changes in building-cmake.md</p>
<ul>
<li>Converted apt-get into apt</li>
<li>Updated zlib download link</li>
<li>Updated the branch for the stable version of OpenSSL (as <a href="https://github.com/john-preston" class="user-mention">@john-preston</a> says in <a href="https://github.com/telegramdesktop/tdesktop/issues/2519" class="issue-link js-issue-link" data-url="https://github.com/telegramdesktop/tdesktop/issues/2519" data-id="182304847" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#2519</a>)</li>
<li>Added ppa for libdrm</li>
<li>Added some troubleshooting for libicu* libraries (as suggested in <a href="https://github.com/telegramdesktop/tdesktop/issues/2719" class="issue-link js-issue-link" data-url="https://github.com/telegramdesktop/tdesktop/issues/2719" data-id="196277327" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#2719</a> )</li>
</ul>
<p>I think we can Close <a href="https://github.com/telegramdesktop/tdesktop/issues/2719" class="issue-link js-issue-link" data-url="https://github.com/telegramdesktop/tdesktop/issues/2719" data-id="196277327" data-error-text="Failed to load issue title" data-permission-text="Issue title is private">#2719</a> with this Pull Request.</p>
<p>Tested in Ubuntu 16.04</p>

<hr>

<h4>You can view, comment on, or merge this pull request online at:</h4>
<p>&nbsp;&nbsp;<a href='https://github.com/telegramdesktop/tdesktop/pull/2926'>https://github.com/telegramdesktop/tdesktop/pull/2926</a></p>

<h4>Commit Summary</h4>
<ul>
  <li>Updated building-cmake.md for Ubuntu 16.04</li>
</ul>

<h4>File Changes</h4>
<ul>
  <li>
    <strong>M</strong>
    <a href="https://github.com/telegramdesktop/tdesktop/pull/2926/files#diff-0">doc/building-cmake.md</a>
    (48)
  </li>
</ul>

<h4>Patch Links:</h4>
<ul>
  <li><a href='https://github.com/telegramdesktop/tdesktop/pull/2926.patch'>https://github.com/telegramdesktop/tdesktop/pull/2926.patch</a></li>
  <li><a href='https://github.com/telegramdesktop/tdesktop/pull/2926.diff'>https://github.com/telegramdesktop/tdesktop/pull/2926.diff</a></li>
</ul>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />You are receiving this because you are subscribed to this thread.<br />Reply to this email directly, <a href="https://github.com/telegramdesktop/tdesktop/pull/2926">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe-auth/AUWsXDjktizwqZSNOQSmggk8Q4cqk29Nks5rV-AKgaJpZM4LuJXq">mute the thread</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/AUWsXHlnAZi_nSyM-m-87cuFsFE2Rclqks5rV-AKgaJpZM4LuJXq.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/telegramdesktop/tdesktop/pull/2926"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

<script type="application/json" data-scope="inboxmarkup">{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/telegramdesktop/tdesktop","title":"telegramdesktop/tdesktop","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/telegramdesktop/tdesktop"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"Updated building-cmake.md for Ubuntu 16.04 (#2926)"}],"action":{"name":

auchri commented:
Hi,
thanks for the pull request!

Please read our [contributing policy](https://github.com/telegramdesktop/tdesktop/blob/master/.github/CONTRIBUTING.md). You'll need to make a pull request with the "Signed-off-by:" signature being the last line of your commit message, like it is described in [sign your work](https://github.com/telegramdesktop/tdesktop/blob/master/.github/CONTRIBUTING.md#sign-your-work) section. That will grant your work into the public domain.

(See travis build)

mymedia2 commented:
Nice job!

But why do you recommend to add dangerous PPA repo? It can break a system as described on [its page](https://launchpad.net/~xorg-edgers/+archive/ubuntu/ppa). Why don't you use normal Ubuntu repositories?

Also you create several, symbolic links in the system directory. This is a bad way. You should not create or modify system files without using the package manager. I think it'd be better to remove absolute paths in [Telegram/gyp/qt.gypi](https://github.com/telegramdesktop/tdesktop/blob/28899a642b98e2bba77a172be83a6cc487c56759/Telegram/gyp/qt.gypi#L135) file instead.

mymedia2 commented on doc/building-cmake.md:
```diff
@@ -187,6 +193,20 @@ In Terminal go to **/home/user/TBuild/tdesktop/Telegram** and run
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicutu.a /usr/lib/libicutu.a
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicui18n.a /usr/lib/libicui18n.a
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicuuc.a /usr/lib/libicuuc.a
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicudata.a /usr/lib/libicudata.a
```
By the way, on x32 computer you have to write `i386-linux-gnu`, rather than `x86_64-linux-gnu`

Helias replied:
Added in brackets a comment that point to put i386-linux-gnu for computer x32, and removed the ppa of libdrm, because on Ubuntu if you install libdrm-dev you get a libdrm with version >= 2.4 (just tested on Ubuntu 14.04)

john-preston commented:
There is a problem with the idea itself.

I need the instructions for the Ubuntu 12.04, not for 16.04 :/ Perhaps those could be some separate doc instructions. I build the production versions on Ubuntu 12.04, because they work after that on Ubuntu 12.04 and above. If I build a version on some new distro it doesn't work on old ones anymore.

Helias commented:
I will test this build in Ubuntu 12.04.5 LTS

Helias commented:
I get this error when I use

```bash
git apply ../../tdesktop/Telegram/Patches/gyp.diff
```

I get these errors

```bash
helias@helias:~/Scrivania/TBuild/Libraries/gyp$ git apply ../../tdesktop/Telegram/Patches/gyp.diff
error: patch failed: pylib/gyp/generator/cmake.py:1070
error: pylib/gyp/generator/cmake.py: patch does not apply
error: patch failed: pylib/gyp/generator/xcode.py:72
error: pylib/gyp/generator/xcode.py: patch does not apply
```

Helias commented:
My Virtualbox crashes and refuse to install again ubuntu 12.04.5 :/
I updated the zlib download link on building-cmake.md, I added a new file building-cmake Ubuntu 14.04+.md and I linked it on the file README.MD

Aokromes commented:
Anyway,
Ubuntu 12.04.5 LTS Precise Pangolin Rel August 7, 2014 April 28, 2017
It will go EOL this month.

Mistero5050 commented:
@Aokromes 

> Anyway,
> Ubuntu 12.04.5 LTS Precise Pangolin Rel August 7, 2014 April 28, 2017
> It will go EOL this month.

Now it's ended

vdmit commented:
+1 for this PR, esp. for these changes:
* libdrm-dev to dependencies;
* proper OpenSSL branch.

john-preston commented on doc/building-cmake_Ubuntu14.04+.md:
```diff
@@ -0,0 +1,212 @@
+##Build instructions for GYP/CMake under Ubuntu 16.04
```
The title says already Ubuntu 16, but all other places indicate Ubuntu 14. Which one is really done here?

john-preston commented on doc/building-cmake_Ubuntu14.04+.md:
```diff
@@ -0,0 +1,212 @@
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicutu.a /usr/lib/libicutu.a
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicui18n.a /usr/lib/libicui18n.a
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicuuc.a /usr/lib/libicuuc.a
+    sudo ln -s /usr/lib/x86_64-linux-gnu/libicudata.a /usr/lib/libicudata.a
```
Perhaps you can suggest a better way (perhaps with modifying the .gyp / .gypi files)? So that it would work both on Ubuntu 12 and Ubuntu 14 (or 16?) and would link static libraries, not .so ones.
